patches/0007-LPS-124728-Avoid-breaking-IE11.patch

```diff
@@ -0,0 +1,63 @@
+From 77e1a8c4a354df7c8edcc550bec7517571b0b275 Mon Sep 17 00:00:00 2001
+From: Julien Castelain <julien.castelain@liferay.com>
+Date: Fri, 18 Dec 2020 13:46:17 +0100
+Subject: [PATCH] LPS-124728 Avoid breaking IE11
+
+IE11 Doesn't support URL, URLSearchParams and Object.entries
+In theory though we have a pollyfill for these but it seems that
+they aren't loaded when getUrl is executed.
+---
+ core/ckeditor_base.js | 35 ++++++++++++++++++++---------------
+ 1 file changed, 20 insertions(+), 15 deletions(-)
+
+diff --git a/core/ckeditor_base.js b/core/ckeditor_base.js
+index 6d9fcbc03..0db736ef1 100644
+--- a/core/ckeditor_base.js
++++ b/core/ckeditor_base.js
+@@ -171,26 +171,31 @@ if ( !window.CKEDITOR ) {
+ 			 * @returns {String} The full URL.
+ 			 */
+ 			getUrl: function( resource ) {
+-				var resourceUrl = resource.startsWith(this.basePath)
+-					? new URL(resource)
+-					: new URL(this.basePath + resource);
++				// If this is not a full or absolute path.
++				if ( resource.indexOf( ':/' ) == -1 && resource.indexOf( '/' ) !== 0 )
++					resource = this.basePath + resource;
+ 
+-				if (!resourceUrl.toString().endsWith('/')) {
+-					if (this.timestamp && !resourceUrl.searchParams.has('t')) {
+-						resourceUrl.searchParams.set('t', this.timestamp);
++				// Add the timestamp, except for directories.
++				if ( this.timestamp && resource.charAt( resource.length - 1 ) != '/' && !( /[&?]t=/ ).test( resource ) )
++					resource += ( resource.indexOf( '?' ) >= 0 ? '&' : '?' ) + 't=' + this.timestamp;
++
++				Object.keys(CKEDITOR.ADDITIONAL_RESOURCE_PARAMS).forEach(function(key) {
++					var value = CKEDITOR.ADDITIONAL_RESOURCE_PARAMS[key];
++
++					if ((resource.charAt(resource.length - 1) === '/') || (/\.css$/).test(resource) && key === 'languageId') {
```

Comment:
Rather than repeating this `charAt()` check for every param, maybe you should do it once, like the old version did.

ie.

```js
if (resource.charAt(resource.length - 1) !== '/') {
    if (this.timestamp && !/[&?]t=/.test(resource)) {
        resource += (resource.indexOf('?') >= 0 ? '&' : '?') + 't=' + this.timestamp;
    }

    Object.keys(CKEDITOR.ADDITIONAL_RESOURCE_PARAMS).forEach(function(key) {
        var value = CKEDITOR.ADDITIONAL_RESOURCE_PARAMS[key];

        if (/\.css$/.test(resource) && key === 'languageId') {
            return;
        }

        var regexp = new RegExp('[&?]' + key);

        if (!regexp.test(resource)) {
            resource += (resource.indexOf('?') >= 0 ? '&' : '?') + key + '=' + value;
        }
    });
}

return resource;
```

It is also tempting to factor out the repeated ternaries... and fix a bug there with that `new RegExp` call which given key "foo" will create a regexp that matches `?foo`, `?foobar`, and `?foobarbaz` etc...

```js
function set(key, value) {
    if (!resource.match(new RegExp('[&?]' + key + '\\b'))) {
        resource += (resource.indexOf('?') >= 0 ? '&' : '?') + key + '=' + value;
    }
}

if (resource.charAt(resource.length - 1) !== '/') {
    if (this.timestamp) {
        set('t', this.timestamp);
    }

    Object.keys(CKEDITOR.ADDITIONAL_RESOURCE_PARAMS).forEach(function(key) {
        var value = CKEDITOR.ADDITIONAL_RESOURCE_PARAMS[key];

        if (!/\.css$/.test(resource) || !key === 'languageId') {
            set(key, value);
        }
    });
}

return resource;
```

Reply:
Thanks. Updated in [a11ce92](https://github.com/liferay/liferay-ckeditor/pull/146/commits/a11ce921e841208128bb7f6511b97455bc9bf2c7)